tests/scripts/test_dev.bash

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+# Before running this script ########################################################
+
+# Set up branch
+# If you want to test `main`, do the following:
+# git fetch upstream main
+# git checkout -b test_main_<date> upstream/main
+# git log # check the commits match https://github.com/E3SM-Project/zppy/commits/main
+
+# Set these parameters:
+
+# Make sure you do not have important changes in any of these directories. This script will reset them!
+ZPPY_DIR=/home/ac.forsyth2/zppy/
+DIAGS_DIR=/home/ac.forsyth2/e3sm_diags/
+ZI_DIR=/home/ac.forsyth2/zppy-interfaces/
+
+ZPPY_DEV=zppy_dev_n516
+DIAGS_DEV=diags_dev_2024_12_13
+ZI_DEV=zi_dev_2024_12_13
+
+DIAGS_ENV_CMD="source /home/ac.forsyth2/miniconda3/etc/profile.d/conda.sh; conda activate ${DIAGS_DEV}"
+ZI_ENV_CMD="source /home/ac.forsyth2/miniconda3/etc/profile.d/conda.sh; conda activate ${ZI_DEV}"
+
+UNIQUE_ID="unique_id"
+
+#####################################################################################
+
+# Set up zppy-interfaces env
+cd ${ZI_DIR}
+git fetch upstream main
+git checkout main upstream/main
+if [ $? != 0 ]; then
+  echo 'ERROR (1): Could not check out zppy-interfaces main branch'
+  exit 1
+fi
+git reset --hard upstream/main
+conda clean --all --y
+# TODO: previous iterations of this script had issues with activating conda environments
```

Comment:
Probably need the `conda` sourcing command too: e.g., `source /gpfs/fs1/home/ac.forsyth2/miniforge3/etc/profile.d/conda.sh`